ci-operator/step-registry/telcov10n/pre-ga-catalog/telcov10n-pre-ga-catalog-commands.sh

```diff
@@ -0,0 +1,136 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ Fix container user ************"
+# Fix user IDs in a container
+[ -e "${HOME}/fix_uid.sh" ] && "${HOME}/fix_uid.sh" || echo "${HOME}/fix_uid.sh was not found" >&2
+
+function wait_until_command_is_ok {
+  cmd=$1 ; shift
+  [ $# -gt 0 ] && sleep_for=${1} && shift && \
+  [ $# -gt 0 ] && max_attempts=${1} && shift
+  [ $# -gt 0 ] && exit_non_ok_message=${1} && shift
+  for ((attempts = 0 ; attempts <  ${max_attempts:=10} ; attempts++)); do
+    echo "Attempting[${attempts}/${max_attempts}]..."
+    set -x
+    eval "${cmd}" && { set +x ; return ; }
+    sleep ${sleep_for:='1m'}
+    set +x
+  done
+
+  echo ${exit_non_ok_message:="[Fail] The exit condition was not met"}
+  if [[ "${TENTATIVE_CREATION}" == "yes" ]] ; then
+    echo "However, since it was set as tentative creation, this failure won't cause the job to stop."
```

Comment:
Why do we need a tentative creation and this message?
It's not so clear to me from the comment to this variable

Reply:
Due to the availability of the catalogs in PreGA and GA versions varies in the OCP versions we want to test, the intention of this change is to provide more flexibility in selecting the catalog to use.

I have sometimes encountered the situation where the OCP version being tested reaches GA and fails to generate the PreGA catalog, but the packages are available in the official repo. So, this step is a best-effort to try to generate the PreGA catalog, but not stop the execution if it does not succeed, in the hope that, when installing the operators, they will be available in the official catalog.

Reply:
Can we always check if GA exists so we can skip PreGA, or is this the preferred method of letting the user decide?

Reply:
It really is a bit like this behavior.

If the user does not define the variable “CATALOGSOURCE_NAME”, the catalog will not be created. However, if it is defined, due to the instability of it, it is allowed to fail. That is the default behavior, but if the variable “TENTATIVE_CREATION” is set to “no” it allows to force the whole job to stop if the creation fails.

With that in mind, the next thing is for the step that installs the operators to have a new “!any” option that indicates to use any available catalog to try to install the packetmanifest of the operator in question. Of course, the “!any” option always tries to install the default catalog (which is GA's) and only searches the rest of the catalogs for the package if it does not find it in GA's catalog. Therefore, the combination of both results in a job that can start to be tested in PreGA, but does not fail if it moves to GA at a certain point in time.

And, of course, this behavior is fully configurable, as I have described.

[Here](https://github.com/openshift/release/pull/52625/commits/57ac8138e9682e3c64773f358011554cde90b30b) you have the whole picture in a commit